Python 3 compatibility

caryan opened the pull request:
Not extensively tested but I seem to be able to go back and forth. Also note new way to get Python 3 atom/enaml stack and JSONLibrary submodule update.

caryan committed:
:art: import ordering and yapf formatting

caryan committed:
Python 3 prints and relative imports

caryan committed:
more python 3 imports and a unicode fix

caryan committed:
Exception.message has been deprecated since Python 2.6

see https://www.python.org/dev/peps/pep-0352/

caryan committed:
more unicode/str fixes

caryan committed:
:memo: update README with Python 3 setup info.

grahamrow commented:
I was able to pip install the dependencies and get it running on OSX for what it's worth: 
e.g. pip3 install git+https://github.com/MatthieuDartiailh/enaml.git@python3-parsing

I'm looking into embedding the quince GUI in PyQLab (or visa versa, what better to serve as the node inspector than PyQLab itself?), but looks like the mismatch in PyQt versions will prohibit this for the moment. Seems like it might happen eventually.

caryan committed:
Merge remote-tracking branch 'origin/develop' into feature/py3-compat

caryan committed:
base64 return bytes in Python 3 so decode to ascii

caryan commented:
@grahamrow 46c3d97 should fix the kernel issues you were seeing

blakejohnson commented:
I'm not sure why this failed unit tests unless it was because conda didn't figure out that it needed to install pyqt. In any case, can we update the Travis script to try both python 2 and python 3 and install the dependencies as you suggest in the new README?

caryan commented:
Yep, I'll tackle that.

Commit:
From bytes to ascii

caryan committed:
Merge pull request #125 from BBN-Q/fix/kernel-encoding

From bytes to ascii

blakejohnson committed:
Fix channel filters in views on py3k.

Should fix #126.

caryan committed:
:art: list comprehension instead of list(filter...)

caryan committed:
allow empty gate channel selection

caryan committed:
restrict gate and triggers to marker channels

caryan committed:
Merge branch 'develop' into feature/py3-compat

caryan commented:
@dieris you've probably given this the best real-world workout. Please merge when you're happy.

Commit:
:fire: unecessary map

Python 3 returns generator so inteded side effect of pop did not occur
but cleaner not to use at all

--CAR and DR

Commit:
updating JSONLibraryUtils -MW

matthewware commented:
I haven't done through testing but this branch seems to be working for me on Maxwell.

matthewware commented:
No I thought you had merged the py3-compat into master.  Seems that wasn't the case.

dieris commented:
I'll do a final test before merging. For some reason, I didn't receive any email notifications on this branch in the last 2 days...

matthewware commented:
I guess I'm still lost...  Wasn't py3-compat on JSONLibraryUtils merged into master?
